protocol/protobuf/url_data.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
```

Comment:
I notice that these protobufs are introduced but are not used anywhere in this PR. Am I correct in thinking that they perhaps belong in another PR?

Reply:
@Samyoul Yes, this PR is a general PR for next 2 PRs